numpy/core/tests/test_multiarray.py

```diff
@@ -35,6 +35,10 @@
 # Need to test an object that does not fully implement math interface
 from datetime import timedelta, datetime
 
+suppress_inexact_match = suppress_warnings()
+suppress_inexact_match.filter(DeprecationWarning,
+                              "inexact matches and case insensitive matches")
```

Comment:
Its OK to use this, but I think nowadays we can just use the pytest warning ignore hooks. I assume those works well. The `suppress_warnings()` solved a real issue, but I think python fixed that mostly by now. Now it would still add better scoping, but we never really need that, so its probably just unnecessary to use this...

Reply:
Thanks for bringing this up @seberg ! Before using suppress_warnings I initially tried `pytest.mark.filterwarnings("ignore::DeprecationWarning")` for example for the _check function in test_conversion_utils but it didnt suppress the error for some reason. Also I saw this decorator sparsely used with the codebase and the places it had used were xfail tests. I didnt dig more into it but switched to suppress_warning. I can look more into this if we are trying to stick to it in the future.

Reply:
Honestly, not sure, as cool as pytest is, I guess its sometimes just too smart for me.